### PR TITLE
Zksync era compatibility

### PR DESCRIPTION
**Pull request by gianbelinche**
# What ❔
This PR modifies lambda adapter to be compatible with zksync era, for that purpose the Storage implementation was re-done.

## Checklist

- [ ] PR title corresponds to the body of PR (we generate changelog entries from PRs).
- [ ] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.
- [ ] Code has been formatted via `cargo fmt` and checked with `cargo clippy`.

### CODE REVIEW

**Review by IAvecilla — APPROVED** [review]
LGTM!

File: compiler_tester/src/vm/eravm/lambda_vm_adapter.rs [hunk 1]
```diff
@@ -98,12 +100,15 @@ pub fn run_vm(
 
         lambda_contract_storage.insert(key, bytecode_u256);
     }
+    let initial_storage = InitialStorageMemory {
+        initial_storage: lambda_storage,
+    };
+    let storage_before_run = initial_storage.clone();
```
Comment (juan518munoz) [inline compiler_tester/src/vm/eravm/lambda_vm_adapter.rs]: Is this variable really needed? Can't we just keep `initial_storage` and `.clone()` it when using for `EraVM::new()`?

**Review by juan518munoz — APPROVED** [review]
LGTM

**Review by Oppen — APPROVED** [review]

File: compiler_tester/src/vm/eravm/lambda_vm_adapter.rs (hunk 1 above)
Comment (gianbelinche) [reply compiler_tester/src/vm/eravm/lambda_vm_adapter.rs]: Done

**Review by fkrause98 — APPROVED** [review]

**Pull request merged**